pkg/util/clusteragent/clusteragent.go

```diff
@@ -242,6 +243,68 @@ func (c *DCAClient) GetNodeLabels(nodeName string) (map[string]string, error) {
 	return labels, err
 }
 
+// GetPodsMetadataForNode queries the datadog cluster agent to get nodeName registered
+// Kubernetes pods metadata.
+func (c *DCAClient) GetPodsMetadataForNode(nodeName string) (apiv1.NamespacesPodsStringsSet, error) {
+	const dcaMetadataPath = "api/v1/tags/pod"
```

Comment:
So you want to reduce the # of calls by using the `api/v1/tags/pod` endpoint of the DCA ?
This endpoint was initially created so you can run `agent metamap <nodename>` in the cluster agent, it is an internall endpoint used for troubleshooting - My main worry is the size of the output collected. 
I think it'd be great to do a bench of this.

Reply:
right, but I don't think that on a give node more than 200 pods is running, and each of them expose more than 10 services. what it gives a moderate message size.

I will create a benchmark to check if it worst to make this change. But one of my concern is to simulate also the possible latency between Agent and Cluster-agent in order to have realistic measurement.

Reply:
After some benchmarks, I can confirm a consequent gain: 
- Queries was done in series
- decrease possible latency between agent and cluster-agent is O(1) instead of O(N) (n = number of pods)